### ISSUE TITLE
gatsby build error: JSON.parse (from json-loader)

### ISSUE THREAD

**Issue by alampros** [gatsbyjs/gatsby#3148]
Occurs when a large number of pages (~20-50) are generated via `gatsby-node.js#createPage`. Works fine when only a handful are generated using the same method.

I've created a repo to reproduce this error: https://github.com/alampros/gatsby-json-parse-bug

## error Building static HTML for pages failed
```
  Error: WebpackError

  - JSON.parse

  - index.js:4 Object.module.exports
    [gatsby-json-parse-bug]/[json-loader]/index.js:4:49

  - From previous event:

  - build-html.js:48 _callee$
    [gatsby-json-parse-bug]/[gatsby]/dist/commands/build-html.js:48:46

  - next_tick.js:131 _combinedTickCallback
    internal/process/next_tick.js:131:7

  - next_tick.js:180 process._tickCallback
    internal/process/next_tick.js:180:9
```

**Comment by attfarhan** [comment]
This sounds like this is the issue I'm seeing in https://github.com/gatsbyjs/gatsby/issues/3139.

**Comment by alampros** [comment]
I'm stumped. The closest I've gotten to narrowing this down is to insert a sleep function into [`develop-html.js` at L21](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/commands/develop-html.js#L21), but even a 200ms sleep still throws the error 50% of the time. 1000ms seems to work for my test scenario, but that's probably because I only have 100 pages.

My gut tells me its something to do with a `debounce`d function or an async `fs.writeFile` call without a missed `await` somewhere, but I can't find anything.

(╯°□°)╯︵ ┻━┻

**Comment by KyleAMathews** [comment]
Should be fixed in #3218 lemme know if you're still seeing troubles!

**Comment by KyleAMathews** [comment]
@alampros thanks so much for your reproduction site! It was perfect for debugging what was going on ❤️
